runs/comparisons/suprb_tuning.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+import click
+import mlflow
+from optuna import Trial
+
+from sklearn.linear_model import Ridge
+from sklearn.utils import Bunch, shuffle
+from sklearn.model_selection import ShuffleSplit
+
+from experiments import Experiment
+from experiments.evaluation import CrossValidate
+from experiments.mlflow import log_experiment
+from experiments.parameter_search import param_space
+from experiments.parameter_search.optuna import OptunaTuner
+from problems import scale_X_y
+
+from suprb import rule, SupRB
+from suprb.logging.combination import CombinedLogger
+from suprb.logging.default import DefaultLogger
+from suprb.logging.stdout import StdoutLogger
+from suprb.optimizer.solution import ga
+from suprb.optimizer.rule import es, origin, mutation
+
+
+random_state = 42
+
+
+def load_dataset(name: str, **kwargs) -> tuple[np.ndarray, np.ndarray]:
+    method_name = f"load_{name}"
+    from problems import datasets
+    if hasattr(datasets, method_name):
+        return getattr(datasets, method_name)(**kwargs)
+
+
+@click.command()
+@click.option('-p', '--problem', type=click.STRING, default='airfoil_self_noise')
+@click.option('-j', '--job_id', type=click.STRING, default='NA')
+def run(problem: str, job_id: str):
+    print(f"Problem is {problem}, with job id {job_id}")
+
+    X, y = load_dataset(name=problem, return_X_y=True)
+    X, y = scale_X_y(X, y)
+    X, y = shuffle(X, y, random_state=random_state)
+
+    estimator = SupRB(
+        rule_generation=es.ES1xLambda(
+            operator='&',
+            n_iter=10_000,
+            init=rule.initialization.MeanInit(fitness=rule.fitness.VolumeWu(),
+                                              model=Ridge(alpha=0.01,
+                                                          random_state=random_state)),
+            mutation=mutation.HalfnormIncrease(),
+            origin_generation=origin.SquaredError(),
+        ),
+        solution_composition=ga.GeneticAlgorithm(n_iter=32, population_size=32),
+        n_iter=32,
+        n_rules=4,
+        verbose=10,
+        logger=CombinedLogger(
+            [('stdout', StdoutLogger()), ('default', DefaultLogger())]),
+    )
+
+    tuning_params = dict(
+        estimator=estimator,
+        random_state=random_state,
+        cv=4,
+        n_jobs_cv=4,
+        n_jobs=4,
+        n_calls=1000,
+        timeout=72 * 60 * 60,  # 72 hours
+        scoring='neg_mean_squared_error',
+        verbose=10
+    )
+
+    @param_space()
+    def suprb_ES_GA_space(trial: Trial, params: Bunch):
+        # ES
+        sigma_space = [0, np.sqrt(X.shape[1])]
+
+        params.rule_generation__mutation__sigma = trial.suggest_float(
+            'rule_generation__mutation__sigma', *sigma_space)
+        params.rule_generation__delay = trial.suggest_int('rule_generation__delay', 10, 100)
```

Comment:
should be conditional on `operator` being '&'. Otherwise, this has no effect and confuses the tuner